src/library/autodj/autodjprocessor.cpp

```diff
@@ -1608,26 +1608,33 @@ void AutoDJProcessor::setTransitionMode(TransitionMode newMode) {
     }
 }
 
+DeckAttributes* AutoDJProcessor::getLeftDeck() {
+    // find first left deck
+    foreach (DeckAttributes* pDeck, m_decks) {
+        if (pDeck->isLeft()) {
+            return pDeck;
+        }
+    }
+    return nullptr;
+}
```

Comment:
and `#include <algorithm>` and then vice-versa for the right equivalent.
```suggestion
DeckAttributes* const AutoDJProcessor::getLeftDeck() {
    const auto candidate = std::find_if(m_decks.cbegin(), m_decks.cend(), [](const auto pDeck){
        return pDeck->isLeft()
    });
    return candidate == m_decks.cend() ? nullptr : candidate;
}
```

Reply:
Also these methods are used a lot. Linear searching vs constant-time indexing is not ideal but manageable I guess as long the number of decks is low.

Reply:
I have decided to use a normal range based loop. This way we get around the temporary "candidate" variable.